### ISSUE TITLE
[RFE] Add coreos.inst.copy_network boolean option

### ISSUE THREAD

**Issue by alonsocamaro** [coreos/coreos-installer#645]
# Feature Request #

The current behavior of the coreos-installer is not to keep the installer's boot time network config. But sometimes that boot time config is actually the desired configuration.

At present when the The boot time network config is required it is required to apply the following **manual procedure** "Using advanced networking options for PXE and ISO installations" in https://docs.openshift.com/container-platform/4.8/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html . In my case, I even cannot use that manually.

This can be greatly simplified by only adding 4 lines of shell code (see at the bottom) in /usr/libexec/coreos-installer-service which adds a coreos.inst.copy_network option.

In my case adding this option also eliminates the need of using https://docs.openshift.com/container-platform/4.8/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html post-install.

## Desired Feature ##

Add the following to /usr/libexec/coreos-installer-service

```
# Copy Network boolean
if karg_bool coreos.inst.copy_network; then
  args+=(“--copy-network”)
fi
```

**Comment by jlebon** [comment]
> At present when the The boot time network config is required it is required to apply the following **manual procedure** "Using advanced networking options for PXE and ISO installations" in [docs.openshift.com/container-platform/4.8/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html](https://docs.openshift.com/container-platform/4.8/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html) . In my case, I even cannot use that manually.

The documentation talks about manually configuring the network via `nmtui` and `nmcli`, but you can also do it automatically and then run `coreos-installer install --copy-network` via the live Ignition config. See https://coreos.github.io/coreos-installer/customizing-install/#custom-coreos-installer-invocation.

> This can be greatly simplified by only adding 4 lines of shell code (see at the bottom) in /usr/libexec/coreos-installer-service which adds a coreos.inst.copy_network option.

We know the approach I suggested above is painful to use and we have tooling planned to make it easier so that you don't have to choose between the `coreos.inst.*` convenience kargs and a fully customized install. Existing kargs will continue to be supported, but it's unlikely we'll add support for more kargs.

**Comment by bgilbert** [comment]
We now have `coreos-installer iso customize --network-keyfile` and `pxe customize --network-keyfile`, which allow embedding a NetworkManager keyfile into live boot media and passing it through all the way to the installed system.  We don't currently have a separate `--copy-network` option to `iso/pxe customize`, but it's possible to write:

```
copy-network: true
```

into a file and pass that file to `iso/pxe customize --installer-config`.  If there's a need for it, we could also add a `--copy-network` option to `iso/pxe customize`.

@alonsocamaro, you haven't said where your systems are getting the network configs that you want to copy with `--copy-network`.  Are they generated by an outside process before the live media is booted?  By something in the live system?  If the former, does `--network-keyfile` meet your needs?

**Comment by bgilbert** [comment]
`coreos-installer iso/pxe customize --network-keyfile` is intended to address this use case, and will be available in OpenShift 4.10.  I'll go ahead and close this.  If that functionality is insufficient, please feel free to reopen with more information about your use case.
